### PR TITLE
AnyScale Endpoints Integration

### PR DESCRIPTION
**Pull request by Anindyadeep**
This PR solves:

- issue #26 
- Supports chat_completion, embedding and fine-tuning
- Also need to change some portion of OpenAIConenctor side, such that it handles some issues related to anyscale (where some arguments passed can not be None like `tools` and `tools_choice`) without compromising with functionalities of OpenAIConnector

### CODE REVIEW

File: prem_utils/models.json [hunk 1]
```diff
@@ -593,6 +593,57 @@
...
+                    "model_type": "text2text",
+                    "context_window": 4096
+                },
+                {
+                    "slug": "codellama/CodeLlama-70b-Instruct-hf",
+                    "model_type": "text2text",
+                    "context_window": 4096
+                },
+                {
+                    "slug": "mistralai/Mistral-7B-Instruct-v0.1",
```
Comment (filopedraz) [inline prem_utils/models.json:621]: Double check if the slug is used in another provider.

File: prem_utils/models.json [hunk 2]
```diff
@@ -593,6 +593,57 @@
                     "coming_soon": true
                 }
             ]
+        },
+        {
+            "provider": "anyscale",
+            "models": [
+                {
+                    "slug": "meta-llama/Llama-2-7b-chat-hf",
```
Comment (filopedraz) [inline prem_utils/models.json:601]: Same here. Double check. Slug should be unique.

**Review by filopedraz — CHANGES_REQUESTED** [review]

File: prem_utils/models.json (hunk 1 above)
Comment (Anindyadeep) [reply prem_utils/models.json]: Yes I added the `anyscale/` prefix

File: prem_utils/models.json (hunk 2 above)
Comment (Anindyadeep) [reply prem_utils/models.json]: Yes I added the `anyscale/` prefix

**Pull request merged**